source/cipher_openssl.c

```diff
@@ -467,6 +467,8 @@ int aws_cryptosdk_sig_sign_start_keygen(
 }
 
 void aws_cryptosdk_sig_abort(struct aws_cryptosdk_sig_ctx *ctx) {
+    AWS_PRECONDITION(!ctx || (aws_cryptosdk_sig_ctx_is_valid(ctx) && ctx->alloc));
```

Comment:
Can upgrade this to ```AWS_MEM_IS_READABLE(alloc)```

Reply:
I think I can also move this check to `aws_cryptosdk_sig_ctx_is_valid`, since all the initialization functions set the context.

Reply:
Good idea